adafruit_displayio_layout/widgets/cartesian.py

```diff
@@ -465,46 +593,31 @@ def update_pointer(self, x: int, y: int) -> None:
         :return: None
         rtype: None
         """
-        local_x = int((x - self._xrange[0]) * self._factorx) + self._nudge_x
-        local_y = (
-            int((self._yrange[0] - y) * self._factory) + self.height + self._nudge_y
-        )
-
-        if local_x >= 0 or local_y <= 100:
-            if self._update_line:
-                self._draw_pointers(local_x, local_y)
-                self._update_line = False
-            else:
-                self._pointer.x = local_x
-                self._pointer.y = local_y
-
-    def _set_plotter_line(self) -> None:
-        self.plot_line_point = []
+        self._add_point(x, y)
+        if not self._pointer:
+            self._draw_pointers(
+                self.plot_line_point[-1][0],
+                self.plot_line_point[-1][1],
+            )
+        else:
+            self._pointer.x = self.plot_line_point[-1][0]
+            self._pointer.y = self.plot_line_point[-1][1]
 
     def update_line(self, x: int, y: int) -> None:
```

Comment:
I think we should rename this to `draw_line()` or `add_line()` instead of `update_line()` It looks like it was a previously existing function but maybe didn't function as expected in the past?

With the new behavior it now adds a new line to the graph each time you call it. Rather than moving the previously created line.

The `update_pointer()` function changes the location of the pointer (i.e. removes it from where it used to be and adds it in the location so that it changes location)

So with the current name `update_line()` there seems to be different behavior between `update_line()` and `update_pointer()` even though they have the same prefix in their name. I think we should either match the behavior, or choose a different name for `update_line()`. I would lean towards the latter because I think it's cool to be able to add multiple lines to the chart. Maybe at some point we could even make possible to add multiple points to the chart as well with a `draw_point()` or `add_point()` function.

Reply:
yeah - i did not touch the *public api*...
and you are right - it did not really work - at least it definitly also did not update the line... it draw a new one each time...
so a api name change at this point makes sens to me!
i like the `add_line()` version - as it most accurately describes what happens internally...
i also added a `clear_lines()` function. that just does what it says ;-)